core/new-gui/src/app/common/service/user/user-dictionary/dictionary.service.ts

```diff
@@ -0,0 +1,518 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { from, Observable, Subject } from 'rxjs';
+import * as Ajv from 'ajv';
+import { AppSettings } from 'src/app/common/app-setting';
+import { JSONSchema7 } from 'json-schema';
+import { catchError, map, shareReplay } from 'rxjs/operators';
+import { isEqual } from 'lodash';
+
+/**
+ * User-Dictionary service stores and retrieves string key-value pairs on a per-user basis
+ * It uses a mysql table on the backend side and requires the client to be logged in
+ * @author Albert Liu
+ */
+
+enum RESPONSE_CODE {
+  CODE_ERROR = -1,
+  CODE_ENTRY = 0,
+  CODE_DICT = 1,
+  CODE_CONFIRM = 2,
+}
+
+enum GET_REQUEST_TYPE {
+  GET_ENTRY = 1,
+  GET_DICT = 2,
+}
+
+// Responses describe the json format of responses returned by the backend
+type ErrorResponse = Readonly<{
+  code: RESPONSE_CODE.CODE_ERROR
+  message: string
+}>;
+
+type EntryResponse = Readonly<{
+  code: RESPONSE_CODE.CODE_ENTRY
+  result: string
+}>;
+
+type DictResponse = Readonly<{
+  code: RESPONSE_CODE.CODE_DICT
+  result: UserDictionary
+}>;
+
+type ConfirmationResponse = Readonly<{
+  code: RESPONSE_CODE.CODE_CONFIRM
+  result: string
+}>;
+
+type UserDictionaryResponse = ErrorResponse | EntryResponse | DictResponse | ConfirmationResponse;
+
+// Requests describe the json format of requests sent to the backend
+type GetEntryRequest = {
+  key: string;
+  requestType: GET_REQUEST_TYPE.GET_ENTRY;
+};
+
+type GetAllRequest = {
+  requestType: GET_REQUEST_TYPE.GET_DICT;
+};
+
+type SetEntryRequest = {
+  key: string,
+  value: string,
+};
+
+type DeleteEntryRequest = {
+  key: string,
+};
+
+const errorResponseSchema: JSONSchema7 = {
+  type: 'object',
+  properties: {
+    code: {const: RESPONSE_CODE.CODE_ERROR},
+    message: {type: 'string'}
+  },
+  additionalProperties: false,
+  required: ['code', 'message']
+};
+
+const entryResponseSchema: JSONSchema7 = {
+  type: 'object',
+  properties: {
+    code: {const: RESPONSE_CODE.CODE_ENTRY},
+    result: {type: 'string'}
+  },
+  additionalProperties: false,
+  required: ['code', 'result']
+};
+
+const dictResponseSchema: JSONSchema7 = {
+  type: 'object',
+  properties: {
+    code: {const: RESPONSE_CODE.CODE_DICT},
+    result: {
+      type: 'object',
+      additionalProperties: {
+        type: 'string'
+      }
+    }
+  },
+  additionalProperties: false,
+  required: ['code', 'result']
+};
+
+const confirmationResponseSchema: JSONSchema7 = {
+  type: 'object',
+  properties: {
+    code: {const: RESPONSE_CODE.CODE_CONFIRM},
+    result: {type: 'string'}
+  },
+  additionalProperties: false,
+  required: ['code', 'result']
+};
+
+export type UserDictionary = {
+  [Key: string]: string;
+};
+
+export enum EVENT_TYPE {
+  GET,
+  SET,
+  DELETE,
+  GET_ALL
+}
```

Comment:
the same comment as backend, it's better to be explicit on the name, `EVENT_TYPE` is too general IMO. Either change the name or you can put this enum inside the service. Same for other type names below as well.